### DISCUSSION TITLE
Seeking a Neovim Plugin for Natural Language Enhancement with Keyboard Shortcut Invocation

### DISCUSSION THREAD

**Discussion by 8ta4** [neovim/neovim#24227] [Q&A]
I'm looking for a tool that can help me improve the naturalness of my English writing within Neovim. I have some essential and some optional features that I'm looking for:

**Essential:**
- The tool should produce natural English, giving me stylistic suggestions and not just grammatical corrections.
- The tool should be easy to use with a keyboard shortcut.

**Optional:**
- The tool should be able to correct selected text. If not, replacing clipboard content is also fine.
- It would be awesome if the tool is free and open-source.

I've tried plugins like `LanguageTool` and `vim-grammarous`, but they seem to be outdated and don't meet all my needs. Besides, vim-grammarous reminds me of my ex. It's glamorous but has a lot of unresolved issues.

Do you have any recommendations for tools or plugins that could help me with this? I'm open to solutions that might not integrate with Neovim directly but can work alongside it.

**Comment by otakutyrant** [comment]
Use [ChatGPT.nvim](https://github.com/jackMort/ChatGPT.nvim)? It can run a prompt like https://github.com/f/awesome-chatgpt-prompts#act-as-an-english-translator-and-improver or `grammar_correction`.

Although I prefer to use https://www.chatbotui.com/ with some custom prompts I used.
